50.005 Computer System Engineering/2018/Networking.md

```diff
@@ -0,0 +1,178 @@
+# Networking
+
+### Table of Contents
+- [Networking](#Networking)
+    - [Table of Contents](#Table-of-Contents)
+  - [The Internet - A nuts & bolts view](#The-Internet---A-nuts--bolts-view)
+  - [The Packet](#The-Packet)
+    - [Single Router Packet Delay Sources:](#Single-Router-Packet-Delay-Sources)
+    - [End-to-end Delay for many hosts](#End-to-end-Delay-for-many-hosts)
+    - [Average Link Utilization](#Average-Link-Utilization)
+    - [Modems](#Modems)
+  - [The Internet - A service view](#The-Internet---A-service-view)
+  - [Internet Protocol Stack (Layering)](#Internet-Protocol-Stack-Layering)
+    - [Internet Control Message Protocol (ICMP)](#Internet-Control-Message-Protocol-ICMP)
+    - [Internet Service Provider (ISP)](#Internet-Service-Provider-ISP)
+  - [Sharing of Information via Switching](#Sharing-of-Information-via-Switching)
+    - [Circuit Switching](#Circuit-Switching)
+    - [Packet Switching](#Packet-Switching)
+
+## The Internet - A nuts & bolts view
+1. Hosts (End Systems) running network apps
+2. Communication Links
+3. Packet Switches
+
+## The Packet
+- Packet = Header + Payload
+    - Header data is used by networking hardware to direct the packet to its destination
+    - Payload is extracted and used by application software.
+
+- Usually, the mode of transmission of packets is:
+`PC -> Switch -> Router -> Modem -> Modem -> Switch -> Servers`
+
+- When the packets arrive at the router, the packet arrival rate to link temporarily exceeds the output link capacity. Hence they would queue in the router buffers, causing a delay.
+    - If there are no free buffers for them to get into queue, then the packets will be dropped. This is known as packet loss.
+
+### Single Router Packet Delay Sources:
+
+![](https://i.imgur.com/V2iaWr9.png)
```

Comment:
For all the diagrams, make sure you have Natalie's blessing to submit here.

Even so, I would like for us to eventually replace these with our own diagrams.

Reply:
Okay!